src/Categories/Category/Monoidal/Instance/Endofunctors.agda

```diff
@@ -0,0 +1,131 @@
+open import Categories.Category.Monoidal using (Monoidal; monoidalHelper)
+open import Categories.Category.Monoidal.Symmetric using (Symmetric; symmetricHelper)
+open import Categories.NaturalTransformation.NaturalIsomorphism using (niHelper)
+
+open import Categories.NaturalTransformation.NaturalIsomorphism
+  using (NaturalIsomorphism; unitorˡ; unitorʳ; associator)
+
+import Categories.Morphism.Reasoning as MR
+
+Endofunctors : Category o ℓ e → Category (o ⊔ ℓ ⊔ e) (o ⊔ ℓ ⊔ e) (o ⊔ e)
+Endofunctors C = Functors C C
+
+module _ where
+  -- Conversion function from natural isomorphisms to isomorphisms in the functor category.
+  ≃→≅ : ∀ {C D : Category o ℓ e}
+      → {F G : Functor C D}
+      → NaturalIsomorphism F G
+      → Categories.Morphism._≅_ (Functors C D) F G
+  ≃→≅ i = record
+    { from = i.F⇒G
+    ; to = i.F⇐G
+    ; iso = record { isoˡ = λ {x} → Iso.isoˡ (i.iso x) ; isoʳ = λ {x} → Iso.isoʳ (i.iso x) }
+    } where module i = NaturalIsomorphism i
+
+module _ where
+  Endofunctors-Monoidal : ∀ {C : Category o ℓ e} → Monoidal (Endofunctors C)
+  Endofunctors-Monoidal {C} = record
+    { ⊗ = record
+      { F₀ = uncurry′ _∘F_
+      ; F₁ = uncurry′ _∘ₕ_
+      ; identity = λ { {F , G} →
+        let module F = Functor F
+            module G = Functor G in
+          begin F.F₁ id ∘ id ≈⟨ F.identity ⟩∘⟨refl ⟩
+                id ∘ id ≈⟨ identity² ⟩
+                id ∎ }
+      ; homomorphism = λ { {F , G} {F₂ , G₂} {F₃ , G₃} {f₁ , f₂} {g₁ , g₂} →
+        let module F₃ = Functor F₃
+            module F₂ = Functor F₂
+            module G = Functor G
+            module G₂ = Functor G₂
+            module f₁ = NaturalTransformation f₁
+            module f₂ = NaturalTransformation f₂
+            module g₁ = NaturalTransformation g₁
+            module g₂ = NaturalTransformation g₂ in
+          begin
+            F₃.₁ (g₂.η _ ∘ f₂.η _) ∘ g₁.η (G.₀ _) ∘ f₁.η (G.₀ _) ≈⟨ pushˡ F₃.homomorphism ⟩
+            F₃.₁ (g₂.η _) ∘ F₃.₁ (f₂.η _) ∘ g₁.η (G.₀ _) ∘ f₁.η (G.₀ _) ≈⟨ refl⟩∘⟨ extendʳ (g₁.sym-commute _) ⟩
+            F₃.₁ (g₂.η _) ∘ g₁.η (G₂.F₀ _) ∘ F₂.F₁ (f₂.η _) ∘ f₁.η (G.₀ _) ≈⟨ sym-assoc ⟩
+            (F₃.₁ (g₂.η _) ∘ g₁.η (G₂.₀ _)) ∘ F₂.₁ (f₂.η _) ∘ f₁.η (G.₀ _) ∎ }
+      ; F-resp-≈ = λ { {_} {E , D} (f≈₁g , f≈₂g) → Functor.F-resp-≈ E f≈₂g ⟩∘⟨ f≈₁g }
+      }
+    ; unit = idF
+    ; unitorˡ = ≃→≅ unitorˡ
+    ; unitorʳ = ≃→≅ unitorʳ
+    ; associator = λ {X} {Y} {Z} → ≃→≅ (associator Z Y X)
+    ; unitorˡ-commute-from = identityˡ
+    ; unitorˡ-commute-to = λ { {f = f} →
+        let module f = NaturalTransformation f in
+        begin (id ∘ f.η _)        ≈⟨ identityˡ ⟩
+              f.η _               ≈⟨ Equiv.sym identityʳ ⟩
+              (f.η _ ∘ id)        ≈⟨ Equiv.sym identityʳ ⟩
+              ((f.η _ ∘ id) ∘ id) ∎ }
+    ; unitorʳ-commute-from = λ { {F} {G} {f = f} →
+        let module G = Functor G in
+        let module f = NaturalTransformation f in
```

Comment:
`f` is an unusual name for a NaturalTransformation.  Also, with `id-comm-sym` you can shorten the proof here as well.